pkg/resources/account_role.go

```diff
@@ -0,0 +1,200 @@
+	"name": {
+		Type:     schema.TypeString,
+		Required: true,
+		// TODO(SNOW-999049): Uncomment once better identifier validation will be implemented
+		// ValidateDiagFunc: IsValidIdentifier[sdk.AccountObjectIdentifier](),
+	},
+	"comment": {
+		Type:     schema.TypeString,
+		Optional: true,
+	},
+	ShowOutputAttributeName: {
+		Type:        schema.TypeList,
+		Computed:    true,
+		Description: "Outputs the result of `SHOW ROLES` for the given role.",
+		Elem: &schema.Resource{
+			Schema: schemas.ShowRoleSchema,
+		},
+	},
+}
+
+func AccountRole() *schema.Resource {
+	return &schema.Resource{
+		Schema: accountRoleSchema,
+
+		CreateContext: CreateAccountRole,
+		ReadContext:   ReadAccountRole,
+		DeleteContext: DeleteAccountRole,
+		UpdateContext: UpdateAccountRole,
+		Description:   "The resource is used for role management, where roles can be assigned privileges and, in turn, granted to users and other roles. When granted to roles they can create hierarchies of privilege structures. For more details, refer to the [official documentation](https://docs.snowflake.com/en/user-guide/security-access-control-overview).",
+
+		CustomizeDiff: ComputedIfAnyAttributeChanged(ShowOutputAttributeName, "name", "comment"),
+
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+	}
+}
+
+func CreateAccountRole(ctx context.Context, d *schema.ResourceData, meta any) diag.Diagnostics {
+	client := meta.(*provider.Context).Client
+
+	id := sdk.NewAccountObjectIdentifier(d.Get("name").(string))
+	req := sdk.NewCreateRoleRequest(id)
+
+	if v, ok := d.GetOk("comment"); ok {
+		req.WithComment(v.(string))
+	}
+
+	err := client.Roles.Create(ctx, req)
+	if err != nil {
+		return diag.Diagnostics{
+			diag.Diagnostic{
+				Severity: diag.Error,
+				Summary:  "Failed to create account role",
+				Detail:   fmt.Sprintf("Account role name: %s, err: %s", id.Name(), err),
+			},
+		}
+	}
+
+	d.SetId(helpers.EncodeSnowflakeID(id))
+
+	return ReadAccountRole(ctx, d, meta)
+}
+
+func ReadAccountRole(ctx context.Context, d *schema.ResourceData, meta any) diag.Diagnostics {
+	client := meta.(*provider.Context).Client
+	id := helpers.DecodeSnowflakeID(d.Id()).(sdk.AccountObjectIdentifier)
+
+	accountRole, err := client.Roles.ShowByID(ctx, id)
+	if err != nil {
+		if errors.Is(err, sdk.ErrObjectNotFound) {
+			d.SetId("")
+			return diag.Diagnostics{
+				diag.Diagnostic{
+					Severity: diag.Warning,
+					Summary:  "Account role not found; marking it as removed",
+					Detail:   fmt.Sprintf("Account role name: %s, err: %s", id.Name(), err),
+				},
+			}
+		}
+		return diag.Diagnostics{
+			diag.Diagnostic{
+				Severity: diag.Error,
+				Summary:  "Failed to show account role by id",
+				Detail:   fmt.Sprintf("Account role name: %s, err: %s", id.Name(), err),
+			},
+		}
+	}
+
+	if err := d.Set("name", sdk.NewAccountObjectIdentifier(accountRole.Name).Name()); err != nil {
+		return diag.Diagnostics{
```

Comment:
why not just `diag.FromErr(err)` (like e.g. in the recently reworked database)?

Reply:
It was this way in the old implementation so I left it because it adds more context than diag.FromErr, but I change it to be consistent with other resources